zcash_primitives/src/note_encryption.rs

```diff
@@ -851,432 +929,729 @@ mod tests {
+        }
     }
 
     #[test]
     fn compact_decryption_with_incorrect_diversifier() {
         let mut rng = OsRng;
+        let height_array = [
+            consensus::MainNetwork::SAPLING_ACTIVATION_HEIGHT,
+            consensus::MainNetwork::CANOPY_ACTIVATION_HEIGHT,
+        ];
 
-        let (ovk, ivk, cv, cmu, epk, mut enc_ciphertext, out_ciphertext) =
-            random_enc_ciphertext(&mut rng);
+        for height_ref in height_array.iter() {
+            let height = *height_ref;
+            let (ovk, ivk, cv, cmu, epk, mut enc_ciphertext, out_ciphertext) =
+                random_enc_ciphertext(height, &mut rng);
 
-        reencrypt_enc_ciphertext(
-            &ovk,
-            &cv,
-            &cmu,
-            &epk,
-            &mut enc_ciphertext,
-            &out_ciphertext,
-            |pt| pt[1..12].copy_from_slice(&find_valid_diversifier().0),
-        );
-        assert_eq!(
-            try_sapling_compact_note_decryption(
-                &ivk,
-                &epk,
+            reencrypt_enc_ciphertext(
+                &ovk,
+                &cv,
                 &cmu,
-                &enc_ciphertext[..COMPACT_NOTE_SIZE]
-            ),
-            None
-        );
+                &epk,
+                &mut enc_ciphertext,
+                &out_ciphertext,
+                |pt| pt[1..12].copy_from_slice(&find_valid_diversifier().0),
+            );
+            assert_eq!(
+                try_sapling_compact_note_decryption(
+                    &consensus::MainNetwork,
+                    height,
+                    &ivk,
+                    &epk,
+                    &cmu,
+                    &enc_ciphertext[..COMPACT_NOTE_SIZE]
+                ),
+                None
+            );
+        }
     }
 
     #[test]
     fn recovery_with_invalid_ovk() {
         let mut rng = OsRng;
+        let height_array = [
+            consensus::MainNetwork::SAPLING_ACTIVATION_HEIGHT,
+            consensus::MainNetwork::CANOPY_ACTIVATION_HEIGHT,
+        ];
 
-        let (mut ovk, _, cv, cmu, epk, enc_ciphertext, out_ciphertext) =
-            random_enc_ciphertext(&mut rng);
+        for height_ref in height_array.iter() {
+            let height = *height_ref;
+            let (mut ovk, _, cv, cmu, epk, enc_ciphertext, out_ciphertext) =
+                random_enc_ciphertext(height, &mut rng);
 
-        ovk.0[0] ^= 0xff;
-        assert_eq!(
-            try_sapling_output_recovery(&ovk, &cv, &cmu, &epk, &enc_ciphertext, &out_ciphertext),
-            None
-        );
+            ovk.0[0] ^= 0xff;
+            assert_eq!(
+                try_sapling_output_recovery(
+                    &consensus::MainNetwork,
+                    height,
+                    &ovk,
+                    &cv,
+                    &cmu,
+                    &epk,
+                    &enc_ciphertext,
+                    &out_ciphertext
+                ),
+                None
+            );
+        }
```

Comment:
The more verbose formatting is unfortunate. Maybe abbreviate `ciphertext` to `ctext`?